src/integrity-checker.js

```diff
@@ -185,7 +185,21 @@ export default class InstallationIntegrityChecker {
       artifacts,
     };
 
-    result.topLevelPatterns = patterns.sort(sortAlpha);
+    result.topLevelPatterns = patterns.sort(sortAlpha).filter(p => {
+      return !workspaceLayout || !workspaceLayout.getManifestByPattern(p);
```

Comment:
Since `workspaceLayout` won't ever change during the lifetime of this function, you can do the following and speed up the cases where workspaces are not used:
```
const topLevelPatterns = workspaceLayout ? patterns.filter(p => !workspaceLayout.getManifestByPattern(p)) : patterns;
result.topLevelPatterns = topLevelPatterns.sort(sortAlpha);
```